builder/evaluator.go

```diff
@@ -173,6 +179,39 @@ func (b *Builder) Run(context io.Reader) (string, error) {
 	return b.image, nil
 }
 
+func (b *Builder) cleanupDockerignore() {
```

Comment:
I'm not a fan of functions like this because it's not clear when and where it should be called as it accepts no input and produces no output.  Maybe this should be part of `readContext` and that should return the Dockerfile ast?

Sorry, I don't mean to lag more on this, we are just dealing with methods like this in another area of the codebase so it's something that I noticed.